### PR TITLE
feat(engagement): manage newsletters subscription lists

### PR DESCRIPTION
**Pull request by miguelpeixe**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Details and instructions at https://github.com/Automattic/newspack-newsletters/pull/873

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/engagement/views/newsletters/index.js [hunk 1]
```diff
@@ -120,20 +124,115 @@ export const NewspackNewsletters = ( { className, onUpdate, isOnboarding = true
 	);
 };
 
-const Newsletters = () => {
-	const [ { status, url, error, newslettersConfig }, updateConfiguration ] = hooks.useObjectState(
-		{}
+export const SubscriptionLists = ( { onUpdate } ) => {
```
Comment (adekbadek) [inline assets/wizards/engagement/views/newsletters/index.js:127]: `onUpdate` prop is not used in any instance, should it be included here?

File: assets/wizards/engagement/views/newsletters/index.js [hunk 2]
```diff
@@ -120,20 +124,115 @@ export const NewspackNewsletters = ( { className, onUpdate, isOnboarding = true
...
+	return (
+		<>
+			<SectionHeader
+				title={ __( 'Subscription Lists', 'newspack' ) }
+				description={ __( 'Manage the lists available for subscription.', 'newspack' ) }
+			/>
+			{ error && (
+				<Notice noticeText={ error.message || __( 'Something went wrong.', 'newspack' ) } isError />
+			) }
+			{ lists.map( ( list, i ) => (
```
Comment (adekbadek) [inline assets/wizards/engagement/views/newsletters/index.js:175]: Suggestion for readability:

```suggestion
			{ lists.map( ( list, index ) => (
```

File: assets/wizards/engagement/views/newsletters/index.js [hunk 3]
```diff
@@ -120,20 +124,115 @@ export const NewspackNewsletters = ( { className, onUpdate, isOnboarding = true
...
+	const [ { newslettersConfig }, updateConfiguration ] = hooks.useObjectState( {} );
+	const [ initialProvider, setInitialProvider ] = useState( '' );
+	const [ lockedLists, setLockedLists ] = useState( false );
+
 	useEffect( () => {
-		fetchJetpackMailchimpStatus().then( updateConfiguration );
-	}, [] );
-	const isConnected = status === 'active';
+		const provider = newslettersConfig?.newspack_newsletters_service_provider;
+		console.log( provider, initialProvider );
```
Comment (adekbadek) [inline assets/wizards/engagement/views/newsletters/index.js:217]:
```suggestion
```

File: assets/wizards/engagement/views/newsletters/index.js [hunk 4]
```diff
@@ -148,40 +247,16 @@ const Newsletters = () => {
 				isOnboarding={ false }
 				onUpdate={ config => updateConfiguration( { newslettersConfig: config } ) }
 			/>
-			<SectionHeader
-				title={ __( 'Signup Forms', 'newspack' ) }
-				description={ () => (
-					<>
-						{ __(
-							'Connects your site to Mailchimp and sets up a Mailchimp block you can use to get new subscribers for your newsletter',
```
Comment (adekbadek) [inline assets/wizards/engagement/views/newsletters/index.js:156]: Along with this removal, the Mailchimp-via-Jetpack connection in the Connections wizard should be removed, too:

image

**Review by adekbadek — CHANGES_REQUESTED** [review]

File: assets/wizards/engagement/views/newsletters/index.js (hunk 1 above)
Comment (miguelpeixe) [reply assets/wizards/engagement/views/newsletters/index.js:127]: It's not currently used but I don't see why not keep it as a component feature.

File: assets/wizards/engagement/views/newsletters/index.js (hunk 2 above)
Comment (miguelpeixe) [reply assets/wizards/engagement/views/newsletters/index.js]: Done in 290c959dec053dd26be8c9d50cf55a2af87e7177

File: assets/wizards/engagement/views/newsletters/index.js (hunk 4 above)
Comment (miguelpeixe) [reply assets/wizards/engagement/views/newsletters/index.js:156]: Done in ff382cb1cebd8f73461f9d204508ddb751937858

**Review by adekbadek — CHANGES_REQUESTED** [review]

File: assets/wizards/engagement/views/newsletters/index.js [hunk 5]
```diff
@@ -120,20 +124,119 @@ export const NewspackNewsletters = ( { className, onUpdate, isOnboarding = true
...
+			.finally( () => setInFlight( false ) );
+	};
+	const handleChange = ( index, name ) => value => {
+		const newLists = [ ...lists ];
+		newLists[ index ][ name ] = value;
+		updateConfig( newLists );
+	};
+	useEffect( fetchLists, [] );
+	if ( ! lists?.length && ! error ) {
+		return null;
```
Comment (adekbadek) [inline assets/wizards/engagement/views/newsletters/index.js:166]: In principle, I think it'd be better to display some UI (loading or error) at all times. But it's not blocking.

**Review by adekbadek — APPROVED** [review]

**Pull request merged**